python/ray/tune/result_grid.py

```diff
@@ -80,6 +81,36 @@ def get_best_result(
             )
         )
 
+    def get_dataframe(
+        self,
+        filter_metric: Optional[str] = None,
+        filter_mode: Optional[str] = None,
+    ) -> pd.DataFrame:
+        """Return dataframe of all trials with their configs and reported results.
+
+        Per default, this returns the last reported results for each trial.
+
+        If ``filter_metric`` and ``filter_mode`` are set, the results from each
+        trial are filtered for this metric and mode. For example, if
+        ``filter_metric="some_metric"`` and ``filter_mode="max"``, for each trial,
+        every received result is checked, and the one where ``some_metric`` is
+        maximal is returned.
+
```

Comment:
Can we provide the schema of the df in doc string?

Reply:
I've added some info in the `Returns` part, but we can't easily provide the full schema as it will depend on the reported metrics (including possibly changing auto-generated metrics).